specs/eip4844/polynomial-commitments.md

```diff
@@ -173,7 +174,9 @@ def hash_to_bls_field(polys: Sequence[Polynomial], comms: Sequence[KZGCommitment
         data += commitment
 
     data += DOMAIN_SEPARATOR_SQUEEZE
-    return bytes_to_bls_field(hash(data))
+    h = hash(data)
+
+    return bytes_to_bls_field(hash(data)), h
```

Comment:
This is equivalent to `return bytes_to_bls_field(h), h`

The rationale here is to initialise the fiat-shamir with the previous challenge that was used. 

I'm not quite sure what is happening with the domain separators, I'd just keep them in the hash_to_bytes_field method to reduce complexity; so domain separators are not in multiple places. Unless the current way is more secure, though I have not seen anything that lends itself to that conclusion

Reply:
I have seen domain separators, where the aggregate kzg protocol is separated from the underlying/original kzg protocol

Reply:
The reason is that `hash_to_bls_field` is called in different contexts. IMO for the domain separators to make sense at all, the interesting part is not what types are being hashed, but what protocol is run. So we can distinguish that via the initializer *if necessary*, although I still don't believe that they are needed so they are all empty strings for now.

The second reason for the `initializer` is that when we run a two stage protocol we need to add the hash of the previous transcript.